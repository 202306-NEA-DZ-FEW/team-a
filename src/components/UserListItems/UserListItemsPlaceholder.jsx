function UserListItemsPlaceholder() {
  return (
    <div className='h-screen w-full items-center flex flex-col gap-4'>
      <div className='w-full flex justify-between mb-4'>
        <div className='w-[300px] h-[30px] rounded-full bg-gray-300 animate-pulse'></div>
        <div className='w-[30px] h-[30px] rounded-full bg-gray-300 animate-pulse'></div>
      </div>
      <div className='w-full h-[90px] rounded-xl bg-gray-300 animate-pulse'></div>
      <div className='w-full h-[90px] rounded-xl bg-gray-300 animate-pulse'></div>
      <div className='w-full h-[90px] rounded-xl bg-gray-300 animate-pulse'></div>
      <div className='w-full h-[90px] rounded-xl bg-gray-300 animate-pulse'></div>
      <div className='w-full h-[90px] rounded-xl bg-gray-300 animate-pulse'></div>
    </div>
  );
}

export default UserListItemsPlaceholder;
