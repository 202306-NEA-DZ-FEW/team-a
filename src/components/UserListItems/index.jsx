import Image from "next/image";
import { useTranslation } from "next-i18next";
import mepi from "public/images/mepi1.png";
import { BsFillPencilFill } from "react-icons/bs";
import { RiDeleteBin6Line } from "react-icons/ri";

function UserListItems() {
  const { t } = useTranslation();
  const items = [
    { id: 1, title: "Produdddddddddd1", image: mepi },
    { id: 2, title: "Product 2", image: mepi },
    { id: 3, title: "Product 3", image: mepi },
    { id: 4, title: "Product 4", image: mepi },
  ];
  return (
    <div className='flex flex-col min-h-[400px] mb-6 gap-4'>
      <h1 className='text-3xl self-center'>{t("dashboard:myItems")} </h1>
      <div
        className='flex justify-center items-center flex-wrap gap-4 
    '
      >
        {items.map((item) => (
          <div
            key={item.id}
            className='flex w-11/12 flex-row gap-4 items-center justify-between rounded-full bg-gray-200 py-1 px-2 '
          >
            <Image
              src={item.image}
              alt={item.title}
              width={60}
              height={60}
              className='object-cover rounded-full'
            />
            <div className='flex-1'>
              <h3 className='text-left text-md text-neutral'>{item.title}</h3>
            </div>
            <div className='mx-2 flex items-center gap-2'>
              <div className='btn bg-transparent hover:-mt-2 hover:bg-transparent py-2 px-2'>
                <BsFillPencilFill className='text-xl text-neutral' />
              </div>
              <div className='btn bg-transparent hover:-mt-2 hover:bg-transparent py-2 px-2'>
                <RiDeleteBin6Line className='text-xl text-error' />
              </div>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
}

export default UserListItems;
