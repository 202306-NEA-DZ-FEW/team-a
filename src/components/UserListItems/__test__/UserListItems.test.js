import renderer from "react-test-renderer";
import UserListItems from "..";

it("renders correctly", () => {
  const tree = renderer.create(<UserListItems />).toJSON();
  expect(tree).toMatchSnapshot();
});
