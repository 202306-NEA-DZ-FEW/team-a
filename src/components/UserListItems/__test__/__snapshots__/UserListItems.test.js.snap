// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col min-h-[400px] mb-6 gap-4"
>
  <h1
    className="text-3xl self-center"
  >
    myItems
     
  </h1>
  <div
    className="flex justify-center items-center flex-wrap gap-4    "
  >
    <div
      className="flex w-11/12 flex-row gap-4 items-center justify-between rounded-full bg-gray-200 py-1 px-2 "
    >
      <img
        alt="Produdddddddddd1"
        className="object-cover rounded-full"
        data-nimg="1"
        decoding="async"
        height={60}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={60}
      />
      <div
        className="flex-1"
      >
        <h3
          className="text-left text-md text-neutral"
        >
          Produdddddddddd1
        </h3>
      </div>
      <div
        className="mx-2 flex items-center gap-2"
      >
        <div
          className="btn bg-transparent hover:-mt-2 hover:bg-transparent py-2 px-2"
        >
          <svg
            className="text-xl text-neutral"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"
            />
          </svg>
        </div>
        <div
          className="btn bg-transparent hover:-mt-2 hover:bg-transparent py-2 px-2"
        >
          <svg
            className="text-xl text-error"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4V2H17V4H22V6H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V6H2V4H7ZM6 6V20H18V6H6ZM9 9H11V17H9V9ZM13 9H15V17H13V9Z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="flex w-11/12 flex-row gap-4 items-center justify-between rounded-full bg-gray-200 py-1 px-2 "
    >
      <img
        alt="Product 2"
        className="object-cover rounded-full"
        data-nimg="1"
        decoding="async"
        height={60}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={60}
      />
      <div
        className="flex-1"
      >
        <h3
          className="text-left text-md text-neutral"
        >
          Product 2
        </h3>
      </div>
      <div
        className="mx-2 flex items-center gap-2"
      >
        <div
          className="btn bg-transparent hover:-mt-2 hover:bg-transparent py-2 px-2"
        >
          <svg
            className="text-xl text-neutral"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"
            />
          </svg>
        </div>
        <div
          className="btn bg-transparent hover:-mt-2 hover:bg-transparent py-2 px-2"
        >
          <svg
            className="text-xl text-error"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4V2H17V4H22V6H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V6H2V4H7ZM6 6V20H18V6H6ZM9 9H11V17H9V9ZM13 9H15V17H13V9Z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="flex w-11/12 flex-row gap-4 items-center justify-between rounded-full bg-gray-200 py-1 px-2 "
    >
      <img
        alt="Product 3"
        className="object-cover rounded-full"
        data-nimg="1"
        decoding="async"
        height={60}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={60}
      />
      <div
        className="flex-1"
      >
        <h3
          className="text-left text-md text-neutral"
        >
          Product 3
        </h3>
      </div>
      <div
        className="mx-2 flex items-center gap-2"
      >
        <div
          className="btn bg-transparent hover:-mt-2 hover:bg-transparent py-2 px-2"
        >
          <svg
            className="text-xl text-neutral"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"
            />
          </svg>
        </div>
        <div
          className="btn bg-transparent hover:-mt-2 hover:bg-transparent py-2 px-2"
        >
          <svg
            className="text-xl text-error"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4V2H17V4H22V6H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V6H2V4H7ZM6 6V20H18V6H6ZM9 9H11V17H9V9ZM13 9H15V17H13V9Z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="flex w-11/12 flex-row gap-4 items-center justify-between rounded-full bg-gray-200 py-1 px-2 "
    >
      <img
        alt="Product 4"
        className="object-cover rounded-full"
        data-nimg="1"
        decoding="async"
        height={60}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={60}
      />
      <div
        className="flex-1"
      >
        <h3
          className="text-left text-md text-neutral"
        >
          Product 4
        </h3>
      </div>
      <div
        className="mx-2 flex items-center gap-2"
      >
        <div
          className="btn bg-transparent hover:-mt-2 hover:bg-transparent py-2 px-2"
        >
          <svg
            className="text-xl text-neutral"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"
            />
          </svg>
        </div>
        <div
          className="btn bg-transparent hover:-mt-2 hover:bg-transparent py-2 px-2"
        >
          <svg
            className="text-xl text-error"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4V2H17V4H22V6H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V6H2V4H7ZM6 6V20H18V6H6ZM9 9H11V17H9V9ZM13 9H15V17H13V9Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
